pkg/sources/syslog/syslog.go

```diff
@@ -59,6 +59,44 @@ func NewSyslog(sourceType sourcespb.SourceType, jobID, sourceID int64, sourceNam
 	}
 }
 
+// Validate validates the configuration of the source.
+func (s *Source) Validate() []error {
+	errors := []error{}
+
+	if s.conn.TlsCert != nilString || s.conn.TlsKey != nilString {
+		if s.conn.TlsCert == nilString || s.conn.TlsKey == nilString {
+			errors = append(errors, fmt.Errorf("tls cert and key must both be set"))
+		}
+		if _, err := tls.LoadX509KeyPair(s.conn.TlsCert, s.conn.TlsKey); err != nil {
+			errors = append(errors, fmt.Errorf("error loading tls cert and key: %s", err))
+		}
+	}
+
+	if s.conn.ListenAddress != nilString {
+		switch s.conn.Protocol {
+		case "tcp":
+			_, err := net.Listen(s.conn.Protocol, s.conn.ListenAddress)
```

Comment:
**Question:** Do we need to close the returned socket?

Reply:
They'll automatically be closed when the program terminates, but it'd be better to do it explicitly.